test/twitter/accounts_test.exs

```diff
@@ -0,0 +1,78 @@
+defmodule Twitter.AccountsTest do
+  use Twitter.DataCase, async: true
+
+  alias Twitter.Accounts
+  alias Twitter.Accounts.User
+
+  describe "create_user/1" do
+    @valid_attrs %{
+      name: "User",
+      username: "eva",
+      password: "secret",
+      email: "blabla"
+    }
+    @invalid_attrs %{}
+
+    test "with valid data inserts user" do
+      assert {:ok, %User{id: id} = user} = Accounts.create_user(@valid_attrs)
+      assert user.name == "User"
+      assert user.username == "eva"
+      assert user.email == "blabla"
+      assert [%User{id: ^id}] = Accounts.list_users()
+    end
+
+    test "with invalid data does not insert user" do
+      assert {:error, _changeset} = Accounts.create_user(@invalid_attrs)
+      assert Accounts.list_users() == []
+    end
+
+    test "enforces unique usernames" do
+      assert {:ok, %User{id: id}} = Accounts.create_user(@valid_attrs)
+      assert {:error, changeset} = Accounts.create_user(@valid_attrs)
+
+      assert %{username: ["has already been taken"]} = errors_on(changeset)
+
+      assert [%User{id: ^id}] = Accounts.list_users()
```

Comment:
unit test should only test one thing at a time. Make this test for the {:error, changeset} instead of {:error, changeset} and a successful create user. This is to say that the last assert is unnecessary as this unit test story is tested in `with valid data inserts user`

Reply:
ok, fixed that